contrib/epee/src/net_ssl.cpp

```diff
@@ -257,8 +257,8 @@ bool is_certificate_allowed(boost::asio::ssl::verify_context &ctx, const ssl_con
       return true;
   }
 
-  // we have no checks to execute
-  return true;
+  // if either checklist is non-empty we must have failed it
+  return ssl_context.allowed_fingerprints.empty() && ssl_context.allowed_certificates.empty();
```

Comment:
Fixes for previous stuff in the same PR should be squashed. We don't want to add known bugs :)
Same for the hex conversion earlier, if you keep it.